src/Control/Session.php

```diff
@@ -321,29 +315,31 @@ public function start(HTTPRequest $request)
                 session_cache_limiter($limiter);
             }
 
-            session_start();
+            $sessionParameters = [
+                "cookie_path" => $path,
+                "cookie_domain" => $domain ?: "",
+                "cookie_secure" => $secure,
+                "cookie_httponly" => true
+            ];
+
+            if ($timeout) {
+                $sessionParameters["cookie_lifetime"] = $timeout;
+            }
+
+            session_start($sessionParameters);
 
             if (isset($_SESSION)) {
                 // Initialise data from session store if present
                 $data = $_SESSION;
                 // Merge in existing in-memory data, taking priority over session store data
                 $this->recursivelyApply((array)$this->data, $data);
-            } else {
-                // Use in-memory data if the session is lazy started
-                $data = $this->data;
+                $this->data = $data;
```

Comment:
Is this change (swapping the order) intentional? Seems like it might be accidental?

Regardless, can we get a comment to describe what it's doing? It's not entirely clear at a glance what it does if session isn't set (vs. the exact same line 4 lines down on L340 but with a blank array).

Reply:
yeah, you can only see what i've done with a comparison against the original. Each if else block sets `$data` and then sets it back to `$this->data = $data` at the end. Instead i'm doing the assignment once

Reply:
I've pushed some changes to resolve this :)